src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -45,16 +82,17 @@ export const getDataSource = async (
     DATA_SOURCE_SAVED_OBJECT_TYPE,
     dataSourceId
   );
+
   return dataSource;
 };
 
 export const getCredential = async (
-  dataSource: SavedObject<DataSourceAttributes>,
+  dataSource: DataSourceAttributes,
   cryptography: CryptographyServiceSetup
 ): Promise<UsernamePasswordTypedContent> => {
-  const { endpoint } = dataSource.attributes!;
+  const { endpoint } = dataSource;
 
-  const { username, password } = dataSource.attributes.auth.credentials!;
+  const { username, password } = dataSource.auth.credentials!;
```

Comment:
Is it possible for `dataSource.auth` not to be a valid object?

Reply:
The schema validation will fail the request